my-app/src/components/Transactions/Transactions.tsx

```diff
@@ -1,5 +1,165 @@
-import React from "react";
+import React, { FC, useState } from "react";
+import {
+  StyledWrapper,
+  SectionHeader,
+  StyledWrapperDetails,
+  StyledWrapperDetailsHeader,
+  WrapperImage,
+  StyledTableCell,
+} from "./Transactions.styles";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TablePagination,
+} from "@mui/material";
+import { useSelector } from "react-redux";
+import { RootState } from "../../store/store";
+import { selectUniqueCategoriesTransactions } from "../../utils/transactionsUtils.ts";
+import { SelectSortBy } from "../SelectSortBy/SelectSortBy.tsx";
+import { InputSearch } from "../InputSearch/InputSearch.tsx";
 
-export const Transactions = () => {
-  return <></>;
+interface TransactionsProps {
+  avatar: string;
+  name: string;
+  category: string;
+  date: string;
+  amount: number;
+  recurring: boolean;
+}
+
+const sortFunctions = {
+  All: (transactions: TransactionsProps[]) => transactions,
+  Highest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => b.amount - a.amount),
+  Lowest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => a.amount - b.amount),
+  Oldest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort(
+      (a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()
+    ),
+  Latest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort(
+      (a, b) => new Date(b.date).getTime() - new Date(a.date).getTime()
+    ),
+  "A to Z": (transactions: TransactionsProps[]) =>
```

Comment:
I'd distinguish filter **names** (keys) vs. filter **names** (values).

So this should become:
```suggestion
  "aToZ": (transactions: TransactionsProps[]) =>
```

And mapping to visible labels would be another object.
Try to avoid ever using keys of objects as direct labels to be displayed.

Reply:
Great idea on grouping the sorting functions in one object tho :-)